src/components/files/cleanup.ts

```diff
@@ -0,0 +1,79 @@
+/*
+Script to handle overall system file clean up:
+- Identifies any files in the "files" folder that are not registered in the
+  database ("file" table) and deletes them. 
+- Removes file records from "file" table that are no longer present in the file
+  system.
+- Deletes file records and files that have been marked "to_be_deleted" (by
+  template actions)
+*/
+
+const BATCH_SIZE = 100 // How many file records to scan at a time
+
+import DBConnect from '../databaseConnect'
+import { DateTime } from 'luxon'
+import fs from 'fs'
+import path from 'path'
+import { crawlFileSystem } from '../utilityFunctions'
+import { deleteFile } from '../files/deleteFiles'
+import { FILES_FOLDER, GENERIC_THUMBNAILS_FOLDER } from '../../constants'
+
+const isManualCleanup: Boolean = process.argv[2] === '--cleanup'
+
+interface FilePathData {
+  id: number
+  filePath: string
+}
+const processMissingFileLinks = async () => {
+  const fileIdsToBeDeleted: number[] = []
+  let offset = 0
+  let filePaths: FilePathData[] = await DBConnect.getFilePaths(BATCH_SIZE, offset)
+
+  while (filePaths.length > 0) {
+    filePaths.forEach(({ id, filePath }) => {
+      if (!fs.existsSync(path.join(FILES_FOLDER, filePath))) fileIdsToBeDeleted.push(id)
+    })
+    offset += BATCH_SIZE
+    filePaths = await DBConnect.getFilePaths(BATCH_SIZE, offset)
+  }
+
+  await DBConnect.deleteMissingFileRecords(fileIdsToBeDeleted)
+  return fileIdsToBeDeleted.length
+}
+
+export const cleanUpFiles = async () => {
+  let filesMissingRecords = 0
+
+  // Check if file in database and delete if not
+  const checkFile = async (filePath: string) => {
```

Comment:
Moved this "checkFile" function *inside* the main cleanup function just so we can keep track of the count of deleted files. "checkFile" itself can't return anything, as it's being called many times by "crawlFileSystem", so needed a way to keep a count outside of the function but within the scope of "cleanUpFiles".